src/goose/synopsis/moderator.py

```diff
@@ -56,48 +64,70 @@ def rewrite(self, exchange: Exchange) -> None:
+                    self.notifier.status("Starting new session, preserving history...")
                 self.originals.extend(exchange.messages)
                 if len(exchange.messages) > 1:
                     # we are resuming an existing session, and need to restore state
+                    if self.notifier:
+                        self.notifier.status("Restoring system state from existing session...")
                     system.restore(exchange.messages)
             else:
+                if self.notifier:
+                    self.notifier.status("Updating conversation history...")
                 self.originals.extend(exchange.messages[1:])
 
             exchange.messages.clear()
+            if self.notifier:
+                self.notifier.status("Generating new synopsis with summary and plan...")
             exchange.add(self.get_synopsis(exchange, summarize=True, plan=True))
         else:
             # We're in the state
             # [synopsis, ..., tool_use, tool_result]
             # and we'll keep going but updated the synopsis
             # [new_synopsis, ..., tool_use, tool_result]
+            if self.notifier:
+                self.notifier.status("Updating synopsis after tool use...")
             exchange.messages[0] = self.get_synopsis(exchange)
 
     def get_synopsis(self, exchange: Exchange, summarize: bool = False, plan: bool = False) -> Message:
         if summarize:
+            if self.notifier:
+                self.notifier.status("Generating conversation summary...")
             self.current_summary = self.summarize(exchange)
 
         if plan:
+            if self.notifier:
+                self.notifier.status("Updating action plan...")
             self.current_plan = self.plan(exchange)
 
         return Message.load("synopsis.md", synopsis=self, system=system)
 
     def summarize(self, exchange: Exchange) -> str:
         message = Message.load("summarize.md", synopsis=self, messages=self.originals, exchange=exchange, system=system)
+        if len(self.originals) < 5:
+            return "\n".join([message.summary for message in self.originals])
         model = os.environ.get("SUMMARIZER", exchange.model)
         new_exchange = exchange.replace(moderator=ContextTruncate(), tools=(), system="", messages=[], model=model)
         new_exchange.add(message)
+        if self.notifier:
+            self.notifier.status("Summarizing context")
         return new_exchange.generate().content[0].text
 
     def plan(self, exchange: Exchange) -> str:
         message = Message.load("plan.md", synopsis=self, messages=self.originals, exchange=exchange, system=system)
         model = os.environ.get("PLANNER", exchange.model)
         new_exchange = exchange.replace(moderator=PassiveModerator(), tools=(), system="", messages=[], model=model)
         new_exchange.add(message)
+        if self.notifier:
+            self.notifier.status("Generating plan with model...")
```

Comment:
might just be a dev thing but i swap models often, add the model name to the output?

Reply:
good idea